labs/bbb-webrtc-sfu/lib/audio/AudioManager.js

```diff
@@ -0,0 +1,137 @@
+/*
+ * Lucas Fialho Zawacki
+ * Paulo Renato Lanzarin
+ * (C) Copyright 2017 Bigbluebutton
+ *
+ */
+
+"use strict";
+
+const BigBlueButtonGW = require('../bbb/pubsub/bbb-gw');
+const Audio = require('./audio');
+const BaseManager = require('../base/BaseManager');
+const C = require('../bbb/messages/Constants');
+const Logger = require('../utils/Logger');
+
+module.exports = class AudioManager extends BaseManager {
+  constructor (connectionChannel, additionalChannels, logPrefix) {
+    super(connectionChannel, additionalChannels, logPrefix);
+    this._meetings = {};
+    this._trackMeetingTermination();
+    this.messageFactory(this._onMessage);
+  }
+
+  _trackMeetingTermination () {
+    switch (C.COMMON_MESSAGE_VERSION) {
+      case "1.x":
+        this._bbbGW.on(C.DICONNECT_ALL_USERS, (payload) => {
+          let meetingId = payload[C.MEETING_ID];
+          this._disconnectAllUsers(meetingId);
+        });
+        break;
+      default:
+        this._bbbGW.on(C.DICONNECT_ALL_USERS_2x, (payload) => {
+          let meetingId = payload[C.MEETING_ID_2x];
+          this._disconnectAllUsers(meetingId);
+        });
+    }
+  }
+
+  _disconnectAllUsers(meetingId) {
+    let sessionId = this._meetings[meetingId];
+    if (typeof sessionId !== 'undefined') {
+      Logger.debug(this._logPrefix, 'Disconnecting all users from', sessionId);
+      delete this._meetings[meetingId];
+      this._stopSession(sessionId);
+    }
+  }
+
+  _onMessage(message) {
+    Logger.debug(this._logPrefix, 'Received message [' + message.id + '] from connection', message.connectionId);
+    let session;
+
+    let sessionId = message.voiceBridge.split('-')[0];
+    let voiceBridge = sessionId;
+    let connectionId = message.connectionId;
+
+    if(this._sessions[sessionId]) {
+      session = this._sessions[sessionId];
+    }
+
+    switch (message.id) {
+      case 'start':
+
+        if (!session) {
+          session = new Audio(this._bbbGW, connectionId, voiceBridge);
+        }
+
+        this._meetings[message.internalMeetingId] = {};
```

Comment:
Is there a reason you initialize as Object and then immediately overwrite with the actual value?

Reply:
Nope, prolly some leftover code. Gotcha.